### DISCUSSION TITLE
Connection

### DISCUSSION THREAD

**Discussion by rmanalu15** [senaranya/HL7#114] [Q&A] [answered]
how do i connect and get hl7 messages from connected devices?

**Comment by senaranya** [comment, marked as answer]
This library can't be used for receiving HL7 messages. You need to use a HL7 listener. You can search for "MLLP" (MLLP is the protocol using which HL7 messages are sent) at packagist.org. These listeners are TCP/IP [socket servers](https://www.php.net/manual/en/sockets.examples.php) with some very basic additional logic, so it's also possible to build one manually without too much effort, e.g. using [ReactPhP](https://reactphp.org/socket/#tcpserver)

**Reply by rmanalu15** [reply]
does this library only send and decode hl7 messages?

**Reply by senaranya** [reply]
That is right, it can be used for parsing or generating HL7 messages and sending those. It doesn't support receiving (yet)

**Reply by rmanalu15** [reply]
Thank you very much sir for the information. I really appreciate this library and hope it can be developed further. this is very helpful
